pandas/io/json/_json.py

```diff
@@ -1301,6 +1304,10 @@ def _try_convert_to_date(self, data):
             return data, False
 
         new_data = data
+
+        if new_data.dtype == "string":
```

Comment:
Should this include `ArrowDtype[string]`?

Reply:
No that shouldn't be necessary, you can't end up with arrow strings in columns that could be dates, that can only happened with the automatic conversion when the infer string option is set